server/artifacts/artifact_server.go

```diff
@@ -504,7 +504,7 @@ func (a *ArtifactServer) getWorkflowAndValidate(ctx context.Context, namespace s
 }
 
 func (a *ArtifactServer) validateAccess(ctx context.Context, wf *wfv1.Workflow) error {
-	allowed, err := auth.CanI(ctx, "get", "workflows", wf.Namespace, wf.Name)
+	allowed, err := auth.CanI(ctx, "get", "workflows", wf.Namespace)
```

Comment:
Maybe you misinterpreted my [previous comment on the issue](https://github.com/argoproj/argo-workflows/issues/12161#issuecomment-2155862968) -- I was saying that `name` should actually be used in `auth.CanI` as it seemed to have been _unintentionally_ removed in [this line](https://github.com/argoproj/argo-workflows/pull/2670/files#diff-10789fdfe9fcd47b01b89c63b756fa92802e56ce0cd545b4df4bce1b2f2e8a07L21) of #2670.

If a `name` is specified, it should be part of the query, which I believe will work with `resourceNames`

Reply:
Oh yes sorry, I misinterpreted what you were saying, I'm reverting those changes !